File: 4 - Replace Conditional Logic with Strategy/C#/MPG.ReplaceConditionalLogicWithStrategy/MPG.ReplaceConditionalLogicWithStrategy.Before/CapitalStrategyRevolver.cs
namespace MPG.ReplaceConditionalLogicWithStrategy.Before
{
    public class CapitalStrategyRevolver : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            return (loan.OutstandingRiskAmount() * loan.Duration() * RiskFactor(loan))
                   + (loan.UnusedRiskAmount() * loan.Duration() * UnusedRiskFactor(loan));
        }
    }
}

File: 5 - Form Template Method/Python/before/capital_strategy_revolver.py
from before.capital_strategy import CapitalStrategy


class CapitalStrategyRevolver(CapitalStrategy):
    def __init__(self):
        CapitalStrategy.__init__(self)

    def capital(self, loan):
        return (loan.outstanding_risk_amount() * self.duration(loan) * self.risk_factor_for(loan)) \
               + (loan.unused_risk_amount() * self.duration(loan) * self.unused_risk_factor_for(loan))
